### DISCUSSION TITLE
InstanceSpecification ("object diagrams") notation?

### DISCUSSION THREAD

**Discussion by ebousse** [gaphor/gaphor#2946] [Ideas]
To illustrate the effect of a use case on a system, or to exemplify a pre-condition, or to "test" a given class diagram, a rather spread use of UML is to create so-called "**object diagrams**" using the concept and notation of _InstanceSpecification_ provided by UML. This allows drawing a set of instances of classes, and the links between these instances.

The section "9.8 Instances" of the the UML 2.5.1 specification describes this part of the language, with this introduction:

> InstanceSpecifications represent instances of Classifiers in a modeled system. They are often used to model example
configurations of instances. They may be partial or complete representations of the instances that they correspond to.

The specification also includes a few examples of such diagrams:

![Capture d’écran du 2023-12-20 16-12-01](https://github.com/gaphor/gaphor/assets/5868014/6a3d6b9e-db4e-4fe2-ad07-df2c3a5c900b)
![Capture d’écran du 2023-12-20 16-14-38](https://github.com/gaphor/gaphor/assets/5868014/167038fa-a468-4b45-8c41-1829bd7c193f)

I personnally make these kinds of diagrams rather extensively in the courses I teach. By any chance, would this notation have its place in Gaphor in some future?

**Comment by danyeaw** [comment]
Hi @ebousse, I think this is possible - it looks very similar to what we have implemented for Internal Block Diagrams in the SysML profile. Would you be interested in helping to implement this enhancement?

**Reply by ebousse** [reply]
> Hi @ebousse, I think this is possible - it looks very similar to what we have implemented for Internal Block Diagrams in the SysML profile. 

Great!

> Would you be interested in helping to implement this enhancement?

I would! Unfortunately I am fully booked with my job at the moment, in addition to being a poor Python dev. But I'll keep the idea in mind :)
